File: ex-04/ex-04_c++/C2.cpp
#include "C2.hpp"

namespace p1
{

  // Constructor
  C2::C2() : a2(0)
  {
  }

  // Destructors
  C2::~C2()
  {
  }

  const void C2::accesar()
  {
    C1 c1;
    c1.setA1(this->a2);
    c1.getA1();
  }

  // Getters & Setters
  void C2::setA2(const int &a2) { this->a2 = a2; }
  const int &C2::getA2() const { return this->a2; }
}

File: ex-04/ex-04_java/p1/C2.java
package p1;

public class C2 {
    
    private int a2;

    public C2(int aa) {
        a2 = aa;
    }

    public void seta2(int aa) {
        a2 = aa;
    }

    public int geta2() {
        return a2;
    }
    
    public void acessar()
    {
        C1 objc1 = new C1(10);
        System.out.println("Valor de obj c1 criado: " + objc1.geta1());
        objc1.seta1(20);
        System.out.println("Valor de obj c1 depois de um set: " + objc1.geta1());
    }
}
